app/models/relationship.rb

```diff
@@ -197,7 +197,7 @@ def self.stringify_resource_pairs(resource_pairs, options = {})
   def grandchild_conditions
     t = self.class.arel_table
     t.grouping(t[:ancestry].matches("#{child_ancestry}/%", nil, true).and(
-                 t[:ancestry].does_not_match("#{child_ancestry}/%", nil, true)))
+                 t[:ancestry].does_not_match("#{child_ancestry}/%/%", nil, true)))
```

Comment:
Should this change maybe live in it's on PR?  I feel like we should vet (with specs?) this where it is being used currently, and then introduce the changes in the above to files with more vetting.

Concerned that the previous uses of `grandchild_rels` were relying on the previous functionality of the `LIKE` for this to work, and now we are not confirming that they work as expected for those use cases.

Reply:
yea. to be fair this is currently just plain broken:

```ruby
     t[:ancestry].matches("#{child_ancestry}/%")
.and(t[:ancestry].does_not_match("#{child_ancestry}/%"))
```

aka `ancestry == 'x' && ancestry != 'x'`

Reply:
and if you read the comment above where it states the sql produced, it states it wants the `'%/%'`

but if we had specs, then this would have never gotten in there

Reply:
> yea. to be fair this is currently just plain broken:

Not arguing that, just suggesting that "future `git-blame`rs" and the like might want to see a PR to point to this change isolated for the fix, with tests that prove it, and not have that PR muddied with a performance fix as well.

Currently, this is one commit and one PR.  I feel like our future selves would find that very confusing.

Reply:
My two cents on a path forward:

- break out the bug fix into a different PR, and add a test around that
- merge that PR, and then integrate that with this one